architecture/deployment/oidc-deployment-and-configuration.md

```diff
@@ -0,0 +1,148 @@
+          "jwks_uri": "https://example.eu.auth0.com/.well-known/jwks.json",
+          "token_endpoint": "https://example.eu.auth0.com/oauth/token"
+      }
+    }
+```
+
+- Keycloak (with a keycloak instance set up on a accessible host which can be
+  resolved via `keycloak` on port `8081`, and a realm created (which the client
+  exists within) called `Datalabs`).
+
+```yaml
+    {
+      "client_id": "{{ oidcProviderClientId }}",
+      "redirect_uri": "https://{{ datalabName }}.{{ domain }}/callback",
+      "response_type": "code",
+      "scope": "openid profile",
+      "authority": "http://keycloak:8081/auth/realms/Datalabs",
+      "automaticSilentRenew": true,
+      "accessTokenExpiringNotificationTime": "600",
+      "filterProtocolClaims": true,
+      "loadUserInfo": true
+    }
+```
+
+Once these are configured within the configmap they will be served up by the
+web-app for the client to use. Depending on whether users can sign up or not,
+they may need to be added manually to the user/account list via the OIDC
+provider.
+
+### Authentication Service Configuration
+
+The other element required to be configured to the OIDC endpoint is the
+authentication service. This service is responsible handling requests around
+user permissions and hence must be able to validate the token which is presented
+by end-users, in order to do this it must be able to see the public key of the
+authentication provider.
+
+This is generally but not always provided in the form of a `well-known`
+endpoint. The application has been configured to allow for this as well as
+directly configuring the endpoints that should be used. Below are the variables
+which should be set as environmental variables in in the [authentication service
+manifest](https://github.com/NERC-CEH/datalab-k8s-manifests/blob/master/templates/datalab/datalab-auth-deployment.template.yml)
+when deploying DataLabs.
+
+The following parameters must be specified regardless of the OIDC provider being
+used;
+
+```bash
+OIDC_PROVIDER_DOMAIN
+# This should be the base URL
+OIDC_PROVIDER_AUDIENCE
+# This will be a value which is custom to the DataLabs deployment and will be used as the
+# audience parameter on internal tokens that the authentication service generates, itNERCDL-733-oidc-documentation
+# is typically something like e.g https://datalab.datalabs.nerc.ac.uk/api
+```
+
+In addition to these if the provider does not offer a
+`${OIDC_PROVIDER_DOMAIN}/.well-known/openid-configuration` endpoint which
+describe the URLs that should be used for things like retrieving JWKs & tokens,
+the following additional parameters must also be specified as environmental
+variables (these endpoints use the `OIDC_PROVIDER_DOMAIN` as a base URL).
+
+```bash
+OIDC_OAUTH_TOKEN_ENDPOINT
+# default: '/oauth/token'
+OIDC_JWKS_ENDPOINT
+# default: '/.well-known/jwks.json'
+```
```

Comment:
I think this would be better displayed in a table with Name, Description and Default Value columns